Modules/Core/Common/include/itkBooleanStdVector.h

```diff
@@ -0,0 +1,43 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkBooleanStdVector_h
+#define itkBooleanStdVector_h
+
+#include "itkIntTypes.h"
+#include <vector>
+
+namespace itk
+{
+
+/** The element type of `BooleanStdVectorType`. Should be treated like a Boolean type. */
+using BooleanType = uint8_t;
+
+static_assert(sizeof(BooleanType) <= sizeof(bool), "itk::BooleanType should not be larger than the built-in C++ bool");
```

Comment:
This could go into a separate (G)Test file.

Reply:
Why must "itk::BooleanType not be larger than the built-in C++ bool"?

Reply:
We don't want to accidentally waste memory.

Reply:
If somehow there were a system where this size check did not hold, yes, you would "waste" a bit of memory.  But the alternative is what?  Using a vector of bool?  No, that would be UB and worse than memory waste.  It's not like you have an option smaller than `uint8_t`.

Reply:
> If somehow there were a system where this size check did not hold

That's very hypothetical, as long as we use `uint8_t`.  This check `sizeof(BooleanType) <= sizeof(bool)` is just meant to ensure that we do not _accidentally_ switch to using a larger integer type.  

If it appears that we would somewhere in the future _want_ to use a larger integer type, we would have to adjust this static_assert, of course. But for now I would rather not waste extra memory on this type.